lib/srv/regular/sshserver.go

```diff
@@ -312,6 +313,11 @@ func (s *Server) GetHostUsers() srv.HostUsers {
 // GetHostSudoers returns the HostSudoers instance being used to manage
 // sudoers file provisioning
 func (s *Server) GetHostSudoers() srv.HostSudoers {
+	// we shouldn't allow modifying sudoers on a proxy server
+	if s.proxyMode {
+		return nil
+	}
```

Comment:
This feels a bit weird given that we catch `s.sudoers == nil` below and sub in a no-op impl, but this effectively cancels any attempt at writing to sudoers without throwing an error which is the behavior we want.

An alternative might be to just expose a `GetProxyMode` function on the `Server` interface, which might be worth doing in a follow up PR. I just wanted `TryWriteSudoersFile` to behave properly regardless of what context it's being called from